proxy/http/HttpSessionManager.cc

```diff
@@ -98,7 +113,10 @@ ServerSessionPool::acquireSession(sockaddr const *addr, INK_MD5 const &hostname_
     // Otherwise we need to scan further matches to match the host name as well.
     // Note we don't have to check the port because it's checked as part of the IP address key.
     if (TS_SERVER_SESSION_SHARING_MATCH_IP != match_style) {
-      while (loc && loc->hostname_hash != hostname_hash) {
+      while (loc) {
+        if (loc->hostname_hash == hostname_hash && match_sni(sm, loc->get_netvc())) {
```

Comment:
The original patch did not have match_sni.   But the code was cut-n-pasted several places.  I'd rather not pollute the caller with more details.  Could change the function name to be less confusion.  Like match_sni_if_needed.